src/main/java/org/springframework/data/couchbase/core/ReactiveFindByQueryOperationSupport.java

```diff
@@ -102,49 +124,61 @@ public ReactiveFindByQueryOperationSupport(final ReactiveCouchbaseTemplate templ
 		@Override
 		public Flux<T> all() {
 			return Flux.defer(() -> {
-				String statement = assembleEntityQuery(false);
-				return template.getCouchbaseClientFactory().getCluster().reactive()
-						.query(statement, query.buildQueryOptions(scanConsistency)).onErrorMap(throwable -> {
-							if (throwable instanceof RuntimeException) {
-								return template.potentiallyConvertRuntimeException((RuntimeException) throwable);
-							} else {
-								return throwable;
-							}
-						}).flatMapMany(ReactiveQueryResult::rowsAsObject).map(row -> {
-							String id = row.getString(TemplateUtils.SELECT_ID);
-							long cas = row.getLong(TemplateUtils.SELECT_CAS);
-							row.removeKey(TemplateUtils.SELECT_ID);
-							row.removeKey(TemplateUtils.SELECT_CAS);
-							return template.support().decodeEntity(id, row.toString(), cas, domainType);
-						});
+				String statement = assembleEntityQuery(false, distinctFields);
+				Mono<ReactiveQueryResult> allResult = this.collection == null
+						? template.getCouchbaseClientFactory().getCluster().reactive().query(statement,
+								query.buildQueryOptions(scanConsistency))
+						: template.getCouchbaseClientFactory().getScope().reactive().query(statement,
+								query.buildQueryOptions(scanConsistency));
+				return allResult.onErrorMap(throwable -> {
+					if (throwable instanceof RuntimeException) {
+						return template.potentiallyConvertRuntimeException((RuntimeException) throwable);
+					} else {
+						return throwable;
+					}
+				}).flatMapMany(ReactiveQueryResult::rowsAsObject).map(row -> {
+					String id = "";
+					long cas = 0;
+					if (distinctFields == null) {
+						id = row.getString(TemplateUtils.SELECT_ID);
+						cas = row.getLong(TemplateUtils.SELECT_CAS);
+						row.removeKey(TemplateUtils.SELECT_ID);
+						row.removeKey(TemplateUtils.SELECT_CAS);
+					}
+					return template.support().decodeEntity(id, row.toString(), cas, returnType);
+				});
```

Comment:
If a collection has been specified, execute on the scope, otherwise execute on the cluster.